tracing/otel/provider.go

```diff
@@ -183,7 +197,10 @@ func (o *OtelProvider) SetupGlobalState(ctx context.Context) (CleanupFunc, error
 //
 // If you're interested in further details, please see the `Tracer.Start` function from the OTEL SDK.
 func (o *OtelProvider) Start(ctx context.Context, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
-	return o.tracerFromContext(ctx).Start(ctx, spanName, opts...)
+	ctx, span := o.tracerFromContext(ctx).Start(ctx, spanName, opts...)
+	log := o.getTraceLogger.LogFromCtx(ctx)
+	ctx = log.WithContext(ctx)
```

Comment:
Technically should add a test to ensure that the context is definitely having the logger injected, but I'll leave it up to your judgement.